include/deal.II/base/cuda.h

```diff
@@ -103,6 +103,39 @@ namespace Utilities
 #endif
     }
 
+    /**
+     * Deleter to be used for `std::unique_ptr` pointing to device memory.
+     */
+    template <typename Number>
+    void
+    delete_device_data(Number *device_ptr) noexcept
+    {
+#ifdef DEAL_II_COMPILER_CUDA_AWARE
+      const cudaError_t error_code = cudaFree(device_ptr);
+      (void)error_code;
+      AssertNothrow(error_code == cudaSuccess,
+                    dealii::ExcCudaError(cudaGetErrorString(error_code)));
+#else
+      (void)device_ptr;
```

Comment:
Can you make this an `Assert(false, ...)`? We shouldn't get here, of course, but it's useful to check :-)